base/hashing.jl

```diff
@@ -55,7 +55,6 @@ end
 
 ## symbol & expression hashing ##
 
-hash(x::Symbol, h::UInt) = 3*object_id(x) - h
```

Comment:
@JeffBezanson: is it possible this line was added for performance since the `hash(x::ANY, h::UInt)` method doesn't specialize because of the `ANY` hint?